client/src/components/AddProduct.js

```diff
@@ -0,0 +1,349 @@
+import React, { Component } from "react";
+import axios from "axios";
+import DateFnsUtils from "@date-io/date-fns";
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker
+} from "@material-ui/pickers";
+import { makeStyles } from "@material-ui/core/styles";
+import {
+  TextField,
+  Button,
+  FormControl,
+  Select,
+  MenuItem,
+  Grid
+} from "@material-ui/core";
+
+export default class AddProduct extends Component {
+  state = {
+    title: "",
+    description: "",
+    imageUrl: [],
+    brand: "",
+    category: "",
+    quantity: 0,
+    price: 0,
+    currency: "",
+    tags: [],
+    company: "",
+    location: {
+      latitude: null,
+      longitude: null
+    },
+    // Must be future date
+    availability: new Date(),
```

Comment:
what do you think about having default date as 'null' for availability and warrantyUntil ? This way if user doesn't choose anything no date is saved instead of 'today's date'